stripe/src/main/java/com/stripe/android/model/parsers/NextActionDataParser.kt

```diff
@@ -37,15 +38,37 @@ internal class NextActionDataParser : ModelJsonParser<StripeIntent.NextActionDat
                 json.has(FIELD_URL) ->
                     StripeIntent.NextActionData.RedirectToUrl(
                         Uri.parse(json.getString(FIELD_URL)),
-                        json.optString(FIELD_RETURN_URL)
+                        json.optString(FIELD_RETURN_URL),
+                        MobileDataParser().parse(json.optJSONObject(FIELD_MOBILE) ?: JSONObject())
                     )
                 else -> null
             }
         }
 
+        internal class MobileDataParser : ModelJsonParser<StripeIntent.NextActionData.RedirectToUrl.MobileData> {
+            override fun parse(json: JSONObject): StripeIntent.NextActionData.RedirectToUrl.MobileData? {
+                val type = StripeIntent.NextActionData.RedirectToUrl.MobileData.Type.fromCode(optString(json, FIELD_TYPE))
+                val obj = type?.let { json.optJSONObject(it.code) }
+                return when (type) {
+                    StripeIntent.NextActionData.RedirectToUrl.MobileData.Type.Alipay -> obj?.let {
```

Comment:
do we need a `Type` enum if we have a sealed class?

Reply:
Definitely don't need it but it felt like a nice way to simplify the parsing (that is the only place it is used). When parsing, we first extract the "type" field and then look for a field with that name to parse. e.g.

```
{
  "type": "some_type_here",
  "some_type_here": {
    ''the_actual_data": "some_value"
  }
}
```
That said, we only have 1 possible type right now so it doesn't provide too much value. We could move it to be only visible in the parsing code or just delete it altogether. Don't feel strongly either way

Reply:
Makes sense, fine with keeping it.

Reply:
moved to parser